examples/systemcalls/systemcalls.c

```diff
@@ -93,6 +170,68 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
+    pid_t pid;
+    int status;
+    //printf("^^^^^PID of the Parent Process = %d^^^^^^^\n", getpid());
+
+    pid = fork();
+
+    //printf("Pid Value = %d\n", pid);
+
+    /*Error*/
+    if(pid == -1)
+    {
+        return false;
+    }
+    else if(pid == 0)
+    {
+        //printf("^^^^^PID of the Child Process = %d^^^^^^^\n", getpid());
+        int file = open(outputfile, O_WRONLY | O_CREAT , 0777);
+        if(file == -1)
+        {
+            exit(-1);
+        }
+
+        //printf("File Descriptor = %d\n", file);
+        int file2 = dup2(file , STDOUT_FILENO);
+
+        if(file2 == -1)
+        {
+            exit(-1);
+        }
+
+        //printf()
+        close(file);
+
+        execv(command[0], command);
+        exit(-1);
+
+        //printf("Child Process Exec Return Value = %d\n", execRetVal);
+        
+    }
+    else
+    {
+        if(waitpid(pid, &status, 0) == -1)
+        {
+            return false;
+        }
+        else
+        {
+            if (WIFEXITED(status) == true)    //returns true if child exited normally
+            {
+                printf("\nChild process terminated normally with exit status %d\n", WEXITSTATUS (status));
+                if(WEXITSTATUS(status) != 0)
+                    return false;
+                else
+                    return true;
+            }
+        }
+    }
```

Comment:
@dhirajbennadi think about how you could avoid duplicating the logic between these two functions.